### PR TITLE
delete! for PriorityQueue

### LINKED ISSUE #343: Implement delete!(pq::PriorityQueue, key)
Hello,

I need to be able to remove a key from a `PriorityQueue`.
I noticed that

```julia
delete!(pq::PriorityQueue, key)
```

is not implemented. Maybe it could be useful for some other users

I haven't look closely at `PriorityQueue` code, so please excuse my ignorance about its internal mechanism.

But I think I need to remove key from `pq.index` `Dict` but also remove a `Pair` from  `pq.xs`

What is your opinion about?

```julia
import Base: delete!
function delete!(pq::PriorityQueue, key)
    idx = pq.index[key]
    delete!(pq.index, key)
    deleteat!(pq.xs, idx)
    pq
end
```

Kind regards

### PR DESCRIPTION
**Pull request by scls19fr**
Closes #343

### CODE REVIEW

File: test/test_priority_queue.jl [hunk 1]
```diff
@@ -202,6 +202,13 @@ import Base.Order.Reverse
             @test dequeue_pair!(pq, 'b') == ('b'=>4)
             @test length(pq) == 3
         end
+
+        @testset "delete!" begin
+            pq = PriorityQueue(Base.Order.Forward, "a"=>2, "b"=>3, "c"=>1)
+            pq_out = delete!(pq, "b")
+            @test pq == pq_out
```
Comment (oxinabox) [inline test/test_priority_queue.jl]: this should be `@test pq === pq_out`, as want to check it is outright the same object referencially.

File: test/test_priority_queue.jl [hunk 2]
```diff
@@ -202,6 +202,13 @@ import Base.Order.Reverse
...
+            @test length(pq) == 2
```
Comment (oxinabox) [inline test/test_priority_queue.jl]: I think this should have a test that is more like 
`@test Set(collect(pq)) == Set("a"=>2, "c"=>1)`

**Comment by oxinabox** [comment]
Wait, Travis is failing? :-(
lets not ship it

**Comment by scls19fr** [comment]
I was fixing this also

**Comment by scls19fr** [comment]
You should set this repository to protect master branch when merging

https://github.com/JuliaCollections/DataStructures.jl/settings/branch_protection_rules/

- Require pull request reviews before merging
When enabled, all commits must be made to a non-protected branch and submitted via a pull request with the required number of approving reviews and no changes requested before it can be merged into a branch that matches this rule.

- Require status checks to pass before merging
  - Travis

**Comment by scls19fr** [comment]
https://github.com/JuliaCollections/DataStructures.jl/settings/branch_protection_rules/new/

**Pull request merged**